src/luxonis_ml/loader/augmentations.py

```diff
@@ -33,82 +49,219 @@ def _parse_cfg(
             )
         else:
             resize = A.Resize(image_size[0], image_size[1])
+        # default transforms always triggers
+        default_transform = A.Compose([resize],
+            bbox_params=A.BboxParams(format="coco", label_fields=["bboxes_classes"]),
+            keypoint_params=A.KeypointParams(
+                format="xy", 
+                label_fields=["keypoints_visibility", "keypoints_classes"],
+                remove_invisible=False
+            )
+        )
 
-        all_augs = [resize]
+        single_image_augs = []
+        multi_image_augs = []
         if augmentations:
             for aug in augmentations:
-                all_augs.append(eval(aug["name"])(**aug.get("params", {})))
-        all_augs.append(ToTensorV2())
-
-        return A.Compose(
-            all_augs,
-            bbox_params=A.BboxParams(format="coco", label_fields=["bbox_classes"]),
+                if aug["name"] in ["Mosaic4"]:
+                    self.is_batched = True
+                    self.batch_size = 4
+                    multi_image_augs.append(eval(aug["name"])(**aug.get("params", {})))
+                else:
+                    single_image_augs.append(eval(aug["name"])(**aug.get("params", {})))
```

Comment:
Assuming we will have multiple multi-image augmentations, should we make some enum for single image augs, multi-image augs, or both?

Reply:
Not sure what you mean, can you axplain a bit more? I know that this `if name in []` is not scalable solution but I didn't have an idea of how to do it otherwise. Maybe we first evaluate the augmentation and then check if is of type BatchBasedTransform (based on the [source code](https://github.com/i-aki-y/albumentations/blob/c137e2b8bb48eb25b8865db80027f16059724db7/albumentations/augmentations/geometric/transforms.py#L1502)). And in this case we could also check [mosaic.n_tiles ](https://github.com/i-aki-y/albumentations/blob/c137e2b8bb48eb25b8865db80027f16059724db7/albumentations/augmentations/geometric/transforms.py#L1548C9-L1548C21) to define self.batch_size although this is the only implemented BatchedTransform so far so not sure if others will also follow this class signature.

Reply:
Yeah I think this solution is fine for now and we could wait to see if we use any other batched transforms first. I originally meant we could define an enum that would match to the augmentation names like

```python
class MultiImageAugmentation(Enum):
    MOSAIC = "Mosaic4"
    ...
```
and then a check like
```python
if aug["name"] in MultiImageAugmentation:
```

If we can check for `BatchBasedTransform` it might be even cleaner